cmd/containerd-shim-runhcs-v1/task_hcs.go

```diff
@@ -1008,3 +1026,72 @@ func (ht *hcsTask) requestUpdateContainer(ctx context.Context, resourcePath stri
 	}
 	return ht.c.Modify(ctx, modification)
 }
+
+// publishNotifications monitors the containers notification queue.
+// Currently only OOM events are published.
+//
+// This MUST be called via a goroutine to wait on a background thread.
+func (ht *hcsTask) publishNotifications(ctx context.Context) {
+	var err error
+	ctx, span := trace.StartSpan(ctx, "hcsTask::publishNotifications")
+	defer span.End()
+	defer func() { oc.SetSpanStatus(span, err) }()
+	span.AddAttributes(trace.StringAttribute(logfields.TaskID, ht.id))
+	entity := log.G(ctx).WithField(logfields.TaskID, ht.id)
+
+	nq, err := ht.c.Notifications()
+	if err == notifications.ErrNotificationsNotSupported {
+		entity.Debug("container does not support notifications")
+		return
+	} else if err != nil {
+		entity.WithError(err).Warning("could not access notification queue")
+		return
+	}
+
+outter:
+	for {
+		select {
+		case <-ctx.Done():
+			break outter
```

Comment:
could we just use `continue` instead of a tag?

Reply:
wouldnt that skip to the next iteration of the outer for loop, vs breaking out entirely?

Reply:
I'm not sure I follow. The goal is to go to the next iteration of the for loop right?

Reply:
The goal for the tag is so we can break out of the select and the for loop to exit the whole function if the ctx is done

Reply:
We can just call `return` with no values in that case

Reply:
That works; I didn't realize labels were bad style

Reply:
Oh duh, I'd thought there was work at the very end outside the for but looking again there isn't

Reply:
That's good with me. They're not bad style really, there's uses, but very few and far between 😆